src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,156 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.ElevatorConstants;
+import frc.robot.util.MotorFactory;
+import lib.ctre_shims.TalonEncoder;
+
+public class Elevator extends SubsystemBase {
+
+  private final WPI_TalonFX m_motor;
+  private final DigitalInput m_bottomLimitSwitch;
+  private final DigitalInput m_topLimitSwitch;
+  private final PIDController m_elevatorPID;
+  private final DutyCycleEncoder m_absoluteSpoolEncoder;     
+  private final TalonEncoder m_elevatorMotorEncoder; 
+  private double m_absEncoderZeroPositionTicks;
+  private double clampLow = -ElevatorConstants.kMotorLimit; 
+  private double clampHigh = ElevatorConstants.kMotorLimit;
+  private boolean m_enabled; 
+
+  public Elevator() {
+    m_motor = MotorFactory.createTalonFX(ElevatorConstants.kMotorPort, Constants.kRioCAN);
+    //m_motor.configForwardLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen, 0);
+    //m_motor.configReverseLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen, 0);
+    m_bottomLimitSwitch = new DigitalInput(ElevatorConstants.kBottomLimitSwitchPort); 
+    m_topLimitSwitch = new DigitalInput(ElevatorConstants.kTopLimitSwitchPort);
```

Comment:
The limit switches are plugged into the talonFX, so I think the declaration would be more like the commented out lines.

Reply:
You would also change getting the limit switch to be talon.getSensorCollection().isFwdLimitSwitchClosed();
See more here: https://v5.docs.ctr-electronics.com/en/stable/ch13_MC.html#limit-switches
And then you can remove all the checks for if a limit switch is triggered to not spin the motors.

Reply:
I have uncommented those lines and have gotten the status of the limit switches using that function. See the functions getBottomLimitSwitch() and getTopLimitSwitch()